### ISSUE TITLE
sclang starts but fails to evaluate 

### ISSUE THREAD

**Issue by smoge** [dathinaios/spacemacs-supercollider#7]
sclang starts normally with `sclang-start`, but I can't evaluate any expression. I'm using all the standard PATHs, so I have no idea what's wrong.

**Comment by dathinaios** [comment]
Hello,

What is your OS and SC version? Is `sclang-mode` active? Can you see all the `sclang-mode` key bindings when you hit `,` or `SPC m`?

**Comment by smoge** [comment]
macOS 10.12, SC 3.8 and master. Yes I can see it, it's just the interpreter that does not react to anything.

**Comment by dathinaios** [comment]
Maybe the SC part has not be linked correctly. Try the following and tell me the results:

1. Make sure the `~/Library/Application Support/SuperCollider` folder exists
2. Press `,!` to reset the configuration file (will ask for confirmation).
3. Check that `spacemacs_conf.yaml` has been created in `~/Library/Application Support/SuperCollider`.

**Comment by smoge** [comment]
I've tried all that and I double-checked

**Comment by smoge** [comment]
Looks like something is broken, `sclang` starts normally, all looks OK, but is unable to interact with emacs anymore after initialization

**Comment by dathinaios** [comment]
Unfortunately I can not reproduce it. My system is identical to yours and it is working fine here..

Could you post all that is printed in the `sclang` post window on initialisation as well as the contents of the `spacemacs_conf.yaml` file?

**Comment by smoge** [comment]
Sure!

```
includePaths:
  - "~/.emacs.d/private/supercollider/sc-emacs"
excludePaths:
    []
postInlineWarnings: false

```

And:

```
compiling class library...
initPassOne started
	NumPrimitives = 711
initPassOne done
	compiling dir: '/Applications/SuperCollider.app/Contents/Resources/SCClassLibrary'
	compiling dir: '/Library/Application Support/SuperCollider/Extensions'
	compiling dir: '/Users/smoge/Library/Application Support/SuperCollider/Extensions'
	compiling dir: '/Users/smoge/.emacs.d/private/supercollider/sc-emacs'
	pass 1 done
	numentries = 1746363 / 30637128 = 0.057
	7386 method selectors, 4148 classes
	method table size 29436152 bytes, big table size 245097024
	Number of Symbols 18800
	Byte Code Size 613575
	compiled 687 files in 0.98 seconds
compile done
Emacs: Initializing lisp interface.
Class tree inited in 0.02 seconds
Cleaning up temp synthdefs...


*** Welcome to SuperCollider 3.8.0. *** For help type C-c C-y.
```

**Comment by dathinaios** [comment]
It looks like everything is configured correctly. Just to be sure, are you starting the server because in your printout you haven't done that. You can do that with `, s s` or `SPC m s s`.

**Comment by smoge** [comment]
Scel usually opens the "workspace" by default, on Linux at least. But it does not open here. Is it a change you did? (The project looks promising BTW!)

**Comment by smoge** [comment]
Nothing happens, with key bindings or from the menu options

**Comment by dathinaios** [comment]
Yes, it should open the workspace. I am running out of ideas I'm afraid... Can you try to manually edit `spacemacs_conf.yaml` to remove the quotes from the path and add your full path (without `~`). Post the resulting file content here to check.

**Comment by smoge** [comment]
nothing...

**Comment by smoge** [comment]
```
includePaths:
  - "~/.emacs.d/private/supercollider/sc-emacs"
excludePaths:
    []
postInlineWarnings: false

```

**Comment by smoge** [comment]
```
includePaths:
  - /Users/smoge/.emacs.d/private/supercollider/sc-emacs
excludePaths:
    []
postInlineWarnings: false
```
[image](https://dl.dropboxusercontent.com/u/5977716/Screen%20Shot%202017-01-11%20at%2011.16.31.png)

**Comment by dathinaios** [comment]
I had a look at you image and I you are not on an `scd` or `sc` file. Just to make sure. The commands only work in these files. Although you are right that it should open a workspace file, did you try to create an `scd` file and run the commands from there?

**Comment by smoge** [comment]
Yes, I did, same thing. Something is broken, it would be good to have more debug info and find out what's wrong.

**Comment by dathinaios** [comment]
I am sorry Bernardo but unfortunately I can not reproduce the issue. I have an identical setup and it is working for me... I will let you know here if I think of something else. We will have to wait to see if this issue appears for other users and identify steps to reproduce it.

If you want to troubleshoot further you can try to create a separate account on your machine with a fresh install of spacemacs and no layers at all to make sure there is no conflict there. In terms of debugging I am quite new to `spacemacs` and `elisp` myself.. Check the `*messages*` buffer to see if you can see any errors there but apart from that you will have to do some research yourself. If you make any progress or you pinpoint the problem please let me know!

**Comment by gzmask** [comment]
@smoge 
I figured out a fix:
I had the same issue. OSX, Spacemacs@master, Supercollider 3.8. SCEL works.
Started SC lang fine.
But this line showed up:
file "Support/SuperCollider/spacemacs_conf.yaml" does not exist.

",ss" which suppose to start server, but nothing happens.

Most likely that the path string  "~/Library/Application Support/SuperCollider/spacemacs_conf.yaml" somehow malfunctions with that space between "Application" and "Support".

So I went ahead and change the strings that contains spacemacs_conf.yaml in funcs.el and packages.el to "~/.emacs.d/private/supercollider/spacemacs_conf.yaml". (3 palces)

Also, remove ~/Library/Application Support/SuperCollider folder as the faulty version generates something there that are being reused.

Problem solve. 

@dathinaios  Let me know if you want to patch this yourself or I can sent in a PR.

**Comment by dathinaios** [comment]
Hello @gzmask ,

You should not remove `~/Library/Application Support/SuperCollider` as this is the main folder for SuperCollider data. In any case it will be recreated automatically when you run SC next time. Your solution might have worked but I would expect that you will have problems down the line. Especially if you try to use the official IDE as well. By the way, did you try all the steps that I recommended to @smoge above?

I will try to have a proper look at these issues when I find some time!

**Comment by KonVas** [comment]
This seems to fix the issue of starting but not evaluating and whatnot.

**Comment by dathinaios** [comment]
@KonVas 

I 've run more test and I am afraid I still can not duplicate these issues.

In detail:

I am running `MacOS Sierra`, `SuperCollider 3.8` **(at path /Applications/SuperCollider.app without the supercollider folder!)** and [this homebrew installed emacs version](https://github.com/railwaycat/homebrew-emacsmacport) with `spacemacs`.

I created a separate user account and tested everything from scratch including using [Emacs for Mac](https://emacsformacosx.com/) as my `emacs`.

Everything worked fine. I just had to run `SuperCollider.app` once so that the `SuperCollider` folder was created in `~/LibraryApplication Support`.

Are you doing anything differently? Does it work in a test user account?

**Comment by magwa101** [comment]
I would really like to see this working as the command line in SuperCollider is painful.

In summary, I have similar problems: OSX 10.13.3, SC 3.9.1, emacs 25.2 (home brew).

I open Document.sc, run sclang-start, I get this warning:
WARNING: Could not open directory: '~/.emacs.d/private/supercollider/sc-emacs' and the window does not respond to input.

If I do a ps, I see this process:
/Applications/SuperCollider.app/Contents/MacOS/sclang -l /Users/magwa101/Library/Application Support/SuperCollider/spacemacs_conf.yaml -iscel

If I start from the command line with -iscel command, I get no prompt.
/Applications/SuperCollider.app/Contents/MacOS/sclang -l /Users/magwa101/Library/Application Support/SuperCollider/spacemacs_conf.yaml -iscel

If I run without -iscel I get a prompt, I can start the server, it seems to be functioning:
/Applications/SuperCollider.app/Contents/MacOS/sclang -l /Users/magwa101/Library/Application Support/SuperCollider/spacemacs_conf.yaml

In both cases I get:
WARNING: Could not open directory: '~/.emacs.d/private/supercollider/sc-emacs'

I removed quotes from spaceemacs_conf.yaml and same error.

Hope this helps.

**Comment by dathinaios** [comment]
@magwa101 

I am using `SC 3.8` at the moment. I downloaded `3.9` to test it and it works fine (language and server) apart from errors regarding plugins, which is to be expected as I think the plugin system has changed. I am still on `Sierra` so maybe that has to something to do with it. Apart from that I really don't know since I have tested everything from scratch as per my previous comment.

I assume you have checked that the `~/.emacs.d/private/supercollider/sc-emacs` folder exists and you have the `SupeCollider.app` in `/Applications` and not `/Applications/Supercollider`.

Please comment on this thread if you manage to resolve or further isolate the issue.
